### PR TITLE
Use FALLBACK_ERROR_FORMAT for handlers that return empty()

### LINKED ISSUE #2658: FALLBACK_ERROR_FORMAT does not work with empty()
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Today I tested a handler which always returns `empty()` (see example below). 
I was quite confused by the format of the errors, as they were all in HTML format, even though the fallback was set to json.
It took me a while to figure out that this was because of the `empty()` statement returned by the function, which somehow hijacks the error format, so that the errors are converted to html. 

I could use the `error_format` from now on for handlers returning `empty()`, but was wondering if this is a feature or a bug.

### Code snippet

```python
from sanic import Sanic, empty
from sanic.exceptions import Unauthorized


app = Sanic(__name__)
app.config.FALLBACK_ERROR_FORMAT = "json"

# Returns error in html format
@app.post("/empty")
async def empty_(request):
    raise Unauthorized("bli bla blub")

    return empty()

# Returns error in json format
@app.post("/nothing")
async def nothing(request):
    raise Unauthorized("bli bla blub")


if __name__ == '__main__':
    app.run("127.0.0.1", port=3000, auto_reload=True, debug=True, dev=True)
```

### How do you run Sanic?

As a script

### Operating System

Ubuntu

### Sanic Version

22.9.1

### PR DESCRIPTION
**Pull request by Tronic**
Remove mapping of empty() to HTML, such that the fallback format will rather be used. Also removes mapping of `file`, `redirect` and other untyped responses.

This has effects on two separate cases:
1. Routes using only the removed types will no longer automatically get any error format
2. Routes using known types (text, json, or html) will now receive corresponding error formats when used with removed types that were conflicting, e.g. `json(data) if data else empty()` will receive json error format.

When there is no format from route, due to nothing found or a conflict of formats, Sanic falls back FALLBACK_ERROR_FORMAT and other methods. Note that #2668 affects how that is done but does not fix the issue below.

Users affected by this (either now or after merged) are advised to either add `accept: application/json` or similar header to their request or to define `error_format="json"` (text, html) on their routes, to get a specific format.

Fix #2658

### CODE REVIEW

**Comment by Tronic** [comment]
This passed all tests, which means that we definitely need more tests. Will do on another PR, please merge this if OK!

File: sanic/errorpages.py [hunk 1]
```diff
@@ -404,16 +404,12 @@ def escape(text):
     v: k for k, v in RENDERERS_BY_CONTENT_TYPE.items()
 }
 
+# Handler source code is checked for which response types it returns with the
+# route error_format="auto" (default) to determine which format to use.
 RESPONSE_MAPPING = {
```
[the remark this answers is not in the document]
Comment (Tronic) [reply sanic/errorpages.py:409]: @prryplatypus Do we? I thought that was only created by json() and not instantiated directly, couldn't find any examples of it being used other than for accessors, and neither CoPilot nor ChatGPT knew anything of it either 🤷

[the remark this answers is not in the document]
Comment (ahopkins) [reply sanic/errorpages.py:409]: It certainly could be created manually the same way that `HTTPResponse` could be.

**Comment by Tronic** [comment]
Linter check is failing on some unchanged files, cannot reproduce locally, all formatted OK here.

**Review by ahopkins — APPROVED** [review]

**Pull request merged**